src/etl/impl/CacheLoader.hpp

```diff
@@ -121,7 +122,7 @@ class CacheLoaderImpl {
                 auto [start, end] = cursor.value();
                 LOG(log_.debug()) << "Starting a cursor: " << ripple::strHex(start);
 
-                while (not token.isStopRequested()) {
+                while (not token.isStopRequested() and not cache_.get().isDisabled()) {
```

Comment:
Here is a edge case: If corruption happens at the last cursor, the setFull will still be called. Some cache functions still can be called : https://github.com/XRPLF/clio/blob/develop/src/data/LedgerCache.cpp#L92

Reply:
Not sure how that would happen if both `update` and `setFull` are checking the `disabled_` state and short circuit? I have added more checks inside of cache member functions but i don't think anything needs changing in CacheLoader, wdyt?

Reply:
I saw u added the disable check for member function. Unfortunately, it will break here https://github.com/XRPLF/clio/blob/82b8316978e2b9133cd0b57d123a7ca9e739ee9c/src/etl/impl/LedgerLoader.hpp#L204
In ledgerLoader, whether the cache is disable or not, it will use it.

Reply:
As we discussed on slack - looks like it should be safe due to the disabled checks inside of update and setFull as well as how user-disabled cache is activated after initial load.